### ISSUE TITLE
User's picture is defined per list, not per robot

### ISSUE THREAD

**Issue by xavierba** [sympa-community/sympa#184]
While playing with the user's picture feature, I noticed an user's picture is defined per list, while I would have expected it to be defined per robot. Why such choice ?

**Comment by ikedas** [comment]
FYI.

Though I don't know why, pictures are defined per list at the time of the first implementation of this feature \[[commit](https://github.com/sympa-community/historic-sympa/commit/ba9a33ab)\].

A reasonable explanation I think out is that, users' display name ("gecos") could be defined per list, and pictures feature followed.

**Comment by xavierba** [comment]
Picture, just like gecos, seems to be user specific, not list specific, and thus should be moved to the robot level, not list level.
Also, while thinking about pictures, it could make sense to be able to get the picture from the authentication source (internal, ldap, sql, etc...).

**Comment by ikedas** [comment]
Sorry for obscurity.  I meant: Subscribers' display name ("gecos") could be defined per list, and pictures feature followed.

**Comment by xavierba** [comment]
Yes, I understand your previous comment. I still think both gecos and pictures should be defined at robot level. Or at least, one should be able to have a default per robot identity, which could possibly be overrided at list level, but what would be the use case to have them defined per list ?

**Comment by ikedas** [comment]
We'd be better to continue summarizing issue here (dev list is crowded).

----

Currently user profile (gecos, picture, ...) can be defined by following ways:
  1. "My preferences" page for gecos.
  2. At the time of subscription by user themselves, for gecos.
  3. At the time of addition by owner, for gecos.  (see also issue #194 )
  4. "Subscriber Options" page, for gecos and picture.

In above, 1. is defined as site-wide default (`user_table`), and 2.-4. is defined by each list (`subscriber_table` for gecos).

I understand the proposal is:
  - to merge both profiles (site-wide and by-list),
  - to make them stored by each robot (by-robot) and
  - to allow only users themselves editing profile.

**Comment by xavierba** [comment]
I'm unsure about site-wide vs by-robot. Is that different ? I mean, is site-wide the same as main robot ? Or is site-wide above the robot level (including main robot) ? I guess there is something I have not properly understood here yet.

If site-wide is above by-robot which is itself above by-list, then user's profile data should be inherited into the level below. I mean, if an user has its gecos/pictures/mail defined either locally or from an external authentication source at the site-wide level, the gecos/pictures/mail should be inherited at the by-robot level, except if it is overrided at the by-robot level. Same for by-list which should inherit from by-robot, except if overrided at the by-list level.

Additionally, but this is a different request, an authentication source should offer the option to also provide the mail, gecos and picture.

**Comment by racke** [comment]
The global table _user_table_ has a _gecos_user_ entry, thus the gecos info applies to all robots.
We don't have a table for the robots - relationship between robot and lists is kept in the list table.
Can't see right now where the list definitions for image and gecos are kept in the db.

For sure, gecos storage in Sympa database is suboptimal.

**Comment by ikedas** [comment]
> Can't see right now where the list definitions for image and gecos are kept in the db.

Currently users' pictures are stored in
_`static_content_path`_`/pictures/`_`listname`_`@`_`robot`_`/`_`hash of email`_`.`_`ext`_
then they are by-list.
